File: javascript/countingSort.js
let array = [];
let n = Math.floor(Math.random() * 10000);

for (let i = 0; i < n; i++) {
    array[i] = Math.floor(Math.random() * 1000);
}

let k = Math.max(...array) + 1;
let B = new Array(n);

console.log("Unsorted Array: " + array);

function countingSort(A, B, k) {
    let C = new Array(k).fill(0);

    for (let j = 0; j < A.length; j++) {
        C[A[j]]++;
    }
    for (let i = 1; i < k; i++) {
        C[i] += C[i - 1];
    }
    for (let j = A.length - 1; j >= 0; j--) {
        B[C[A[j]] - 1] = A[j];
        C[A[j]]--;
    }
}

let startTime = performance.now();
countingSort(array, B, k);
let endTime = performance.now();

console.log("Sorted Array: " + B);
console.log("Execution Time: " + (endTime - startTime) + " ms!");

File: python/countingSort.py
import time
import random
import math

def countingSort(A, k):

	C = [0] * (k + 1)
	B = [0] * len(A)

	for i in range(1, len(A)):
		C[A[i]] += 1

	for i in range(1, k + 1):
		C[i] += C[i - 1]

	for j in range(len(A) - 1, -1, -1):
		B[C[A[j]]] = A[j]
		C[A[j]] -= 1

	return B


A = [random.randint(1, 100) for _ in range(10)]
k = max(A)

print("Unsorted array:", A)

startTime = time.time()

sortedArray = countingSort(A, k)

endTime = time.time()
executionTime = (endTime - startTime)

print("Sorted array:", sortedArray	)
print(f"Execution time: {executionTime:.6f} seconds")